controllers/volsync/vshandler.go

```diff
@@ -1253,12 +1253,14 @@ func (v *VSHandler) validateAndProtectSnapshot(
 
 	// Add ownerRef on snapshot pointing to the vrg - if/when the VRG gets cleaned up, then GC can cleanup the snap
 	// Add label to indicate that VolSync should not delete/cleanup this snapshot
-	err = util.NewResourceUpdater(volSnap).
-		AddOwner(v.owner, v.client.Scheme()).
-		AddLabel(VolSyncDoNotDeleteLabel, VolSyncDoNotDeleteLabelVal).
-		Update(v.ctx, v.client)
-	if err != nil {
-		return nil, fmt.Errorf("failed to add owner/label to snapshot %s (%w)", volSnap.GetName(), err)
+	if !v.vrgInAdminNamespace {
+		err = util.NewResourceUpdater(volSnap).
+			AddOwner(v.owner, v.client.Scheme()).
+			AddLabel(VolSyncDoNotDeleteLabel, VolSyncDoNotDeleteLabelVal).
```

Comment:
I think we still need the label added. Otherwise we might clean it up when we shouldn't?

Reply:
@raghavendra-talur Done